src/lib.rs

```diff
@@ -10,7 +10,61 @@ pub mod socket;
 mod topic;
 
 pub use serde_json::Value;
+use tokio::sync::broadcast;
+use tokio::time::error::Elapsed;
 
 pub use self::channel::*;
 pub use self::message::{Event, Payload, PhoenixEvent};
-pub use self::socket::{ConnectError, Socket, SocketError};
+pub use self::socket::{ConnectError, Socket, SpawnError};
+
+#[derive(thiserror::Error, Debug)]
```

Comment:
It's more idiomatic to use thiserror::Error and to derive Error. There is no naming conflict for this

Reply:
Fixed